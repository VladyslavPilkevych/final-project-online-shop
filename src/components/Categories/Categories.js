import React, { useEffect, memo } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { NavLink } from 'react-router-dom';
import Slider from 'react-slick';
import PropTypes from 'prop-types';
import CardItem from '../CardItem/CardItem';
import { filterCategory } from '../../store/actionCreators/filterAC';

import styles from './Categories.module.scss';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';

function SampleNextArrow(props) {
  const { className, style, onClick } = props;
  return (
    <div
      className={className}
      role="button"
      tabIndex={0}
      style={{
        ...style,
        display: 'block',
        background: '#252931',
        opacity: '0.5',
        borderRadius: '30px 0px 0px 30px',
        right: '0',
        zIndex: '10',
        width: '20px',
        height: '32px',
        padding: '13px 0px 1px 11px',
      }}
      onClick={onClick}
    />
  );
}

function SamplePrevArrow(props) {
  const { className, style, onClick } = props;
  return (
    <div
      className={className}
      role="button"
      tabIndex={0}
      style={{
        ...style,
        display: 'block',
        background: '#252931',
        opacity: '0.5',
        borderRadius: '0px 30px 30px 0px',
        left: '0',
        zIndex: '10',
        width: '32px',
        height: '32px',
        paddingTop: '12px',
      }}
      onClick={onClick}
    />
  );
}
function Categories(props) {
  const { productsCategories, imageSrc } = props;
  const allProducts = useSelector((state) => state.products.products);
  const dispatch = useDispatch();
  console.log(allProducts);
  const eachCategory = allProducts.map((product) => {
    if (product.categories === productsCategories) {
      return product;
    }
    return null;
  });
  const resultProducts = eachCategory.filter((element) => element !== null);
  console.log(resultProducts);
  useEffect(() => {
  }, [allProducts]);
  const settings = {
    infinite: true,
    swipeToSlide: true,
    slidesToShow: 5,
    slidesToScroll: 1,
    // overflow: true,
    responsive: [
      {
        breakpoint: 1398,
        settings: {
          slidesToShow: 4,
          slidesToScroll: 1,
        },
      },
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 1,
        },
      },
      {
        breakpoint: 768,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 1,
        },
      },
      {
        breakpoint: 376,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
    // slidesToShow: 1,
    // slidesToScroll: 1,
    prevArrow: <SamplePrevArrow />,
    nextArrow: <SampleNextArrow />,
  };
  return (
    <div className={styles.categoriesWrapper}>
      <div className={styles.categories}>
        <div className={styles.categorieImg} style={{ 'background-image': `url(${imageSrc})` }}>
          <h3>{productsCategories}</h3>
          <div
            onClick={() => {
              dispatch(filterCategory(productsCategories));
            }}
            role="button"
            tabIndex={0}
          >
            <NavLink to={`/filter/${productsCategories}`}>
              {/* <NavLink to="/filter"> */}
              See All Products
            </NavLink>
          </div>
        </div>
        {/* <div className={styles.categorieSlider}> */}
        <div className={styles.categoriesSliderWrapper}>
          <Slider {...settings}>
            {resultProducts && resultProducts.map((elem) => (
              <CardItem
                className={styles.cardItemWrapper}
                itemNo={elem.itemNo}
                name={elem.name}
                // eslint-disable-next-line no-underscore-dangle
                id={elem._id}
                // eslint-disable-next-line no-underscore-dangle
                img={elem.imageUrls[0]}
                // eslint-disable-next-line no-underscore-dangle
                key={elem._id}
                currentPrice={elem.currentPrice}
                quantity={50}
                model={elem.model}
              />
            ))}
          </Slider>
        </div>
      </div>
    </div>
  );
}

SampleNextArrow.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  onClick: PropTypes.func,
};

SampleNextArrow.defaultProps = {
  className: '',
  style: {},
  onClick: () => { },
};
SamplePrevArrow.propTypes = {
  className: PropTypes.string,
  style: PropTypes.object,
  onClick: PropTypes.func,
};

SamplePrevArrow.defaultProps = {
  className: '',
  style: {},
  onClick: () => { },
};

// CategoryProductsContainer.propTypes = {
//   productsCategories: PropTypes.string,
// };

// CategoryProductsContainer.defaultProps = {
//   productsCategories: '',
// };

Categories.propTypes = {
  productsCategories: PropTypes.string,
  imageSrc: PropTypes.string,
};

Categories.defaultProps = {
  productsCategories: '',
  imageSrc: '',
};

export default memo(Categories);