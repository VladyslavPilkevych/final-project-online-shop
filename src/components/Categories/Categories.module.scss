@import '../../assets/styles/global.module.scss';
@import '../../assets/styles/mixins';

.categoriesWrapper {
    max-width: 1398px;
    margin: 0 auto;
}
.categories {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 25px;
}
.categories{
    width: 100%;
}
.categorieImg {
    width: 17%;
    background-repeat: no-repeat;
    background-size: cover;
    text-align: center;
    }
.categorieImg h3 {
    text-transform: capitalize;
    margin-top: 70%;
    @include fonts("Poppins", 28px, white, 600);
}
.categorieImg div {
    margin-top: 47%;
    margin-bottom: 30px;
    cursor: pointer;
    text-decoration: underline;
}
.categorieImg div a {
    @include fonts("Poppins", 13px, white, 400);
}

.categoriesSliderWrapper {
    width: 83%;
}

@media (max-width: 1024px) {
    .categorieImg {
        width: 20%;
    }
    .categorieImg h3 {
        font-size: 24px;
    }
    .categoriesSliderWrapper {
        width: 80%;
    }
}
@media (max-width: 768px) {
    .categorieImg {
        width: 25%;
    }
    .categorieImg h3 {
        font-size: 20px;
    }
    .categoriesSliderWrapper {
        width: 75%;
    }
}
@media (max-width: 550px) {
    .categorieImg h3 {
        font-size: 16px;
    }
}
@media (max-width: 425px) {
    .categories {
        display: block;
        .categorieImg {
            width: 100%;
            height: 82px;
            background-attachment: fixed;
            h3 {
                margin: 0;
                padding-top: 15px;
            }
            div a {
                margin-top: 10px;
            }
        }
        .categoriesSliderWrapper {
            margin: 0 auto;
        }
    }
}
