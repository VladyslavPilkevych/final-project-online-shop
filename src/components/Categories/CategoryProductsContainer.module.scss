.categoriesWrapper {
    max-width: 1165px;
    margin: 0 auto;
    // position: absolute;
    // right: 0;
    // overflow: visible;
}