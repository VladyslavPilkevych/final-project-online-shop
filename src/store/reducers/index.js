import { combineReducers } from 'redux';
import preloaderReducer from './preloaderReducer';
import productsReducer from './productsReducer';
import menuReducer from './menuReducer';
import userReducer from './userReducer';
import searchReducer from './searchReducer';
import cartReducer from './cartReducer';
import miniMenuReducer from './miniMenuReducer';
import filtersCategoriesReducer from './filtersCategoriesReducer';

const reducer = combineReducers({
  loader: preloaderReducer,
  products: productsReducer,
  menu: menuReducer,
  miniMenu: miniMenuReducer,
  filtersCategories: filtersCategoriesReducer,
  user: userReducer,
  search: searchReducer,
  cart: cartReducer,
});

export default reducer;
