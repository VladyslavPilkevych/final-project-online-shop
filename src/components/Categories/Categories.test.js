import { render } from '@testing-library/react';
import Categories from './Categories';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store';

function Component() {
  return (
    <BrowserRouter>
      <Provider store={store}>
        <Categories />
      </Provider>
    </BrowserRouter>
  );
}

describe('Snapshot Categories should render', () => {
  test('Snapshot Categories is rendered', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});