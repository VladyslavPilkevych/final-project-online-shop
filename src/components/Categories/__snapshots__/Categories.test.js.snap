// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Categories should render Snapshot Categories is rendered 1`] = `
<DocumentFragment>
  <div
    class="categoriesWrapper"
  >
    <div
      class="categories"
    >
      <div
        class="categorieImg"
        style="background-image: url();"
      >
        <h3 />
        <div
          role="button"
          tabindex="0"
        >
          <a
            class=""
            href="/filter/"
          >
            See All Products
          </a>
        </div>
      </div>
      <div
        class="categoriesSliderWrapper"
      >
        <div
          class="slick-slider slick-initialized"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
